src/core/components/dag.js

```diff
@@ -5,16 +5,22 @@ const CID = require('cids')
 const pull = require('pull-stream')
 const mapAsync = require('async/map')
 const flattenDeep = require('lodash/flattenDeep')
+const isPlainObject = require('lodash/isPlainObject')
 
 module.exports = function dag (self) {
   return {
     put: promisify((dagNode, options, callback) => {
       if (typeof options === 'function') {
         callback = options
-      } else if (options.cid && (options.format || options.hashAlg)) {
-        return callback(new Error('Can\'t put dag node. Please provide either `cid` OR `format` and `hashAlg` options.'))
-      } else if ((options.format && !options.hashAlg) || (!options.format && options.hashAlg)) {
-        return callback(new Error('Can\'t put dag node. Please provide `format` AND `hashAlg` options.'))
+        options = {}
+      } else if (isPlainObject(options)) {
```

Comment:
How about instead of introduce `isPlainObject()` we'd just introduce a check on whether `options` is defined:

```
if (options && options.cid && (options.format || options.hasAlg)) {
  ...
} else if (/* same thing here */) 
```

And then we'd remove the `else` block and always do sth. like `options = options || {}`